include/boost/geometry/algorithms/detail/overlay/discard_duplicate_turns.hpp

```diff
@@ -88,72 +88,154 @@ inline bool corresponding_turn(Turn const& turn, Turn const& start_turn,
     return count == 2;
 }
 
-template <typename Turns, typename Geometry0, typename Geometry1>
-inline void discard_duplicate_start_turns(Turns& turns,
-                                          Geometry0 const& geometry0,
-                                          Geometry1 const& geometry1)
+//  Verify turns (other than start, and cross) if they are present in the map, and if so,
+//  if they have the other turns as corresponding, discard the start turn.
+template <typename Turns, typename TurnBySegmentMap, typename Geometry0, typename Geometry1>
+void discard_duplicate_start_turns(Turns& turns,
+                                   TurnBySegmentMap const& start_turns_by_segment,
+                                   Geometry0 const& geometry0,
+                                   Geometry1 const& geometry1)
 {
-    // Start turns are generated, in case the previous turn is missed.
-    // But often it is not missed, and then it should be deleted.
-    // This is how it can be
-    // (in float, collinear, points far apart due to floating point precision)
-    //   [m, i s:0, v:6 1/1 (1) // u s:1, v:5  pnt (2.54044, 3.12623)]
-    //   [s, i s:0, v:7 0/1 (0) // u s:1, v:5  pnt (2.70711, 3.29289)]
-
     using multi_and_ring_id_type = std::pair<signed_size_type, signed_size_type>;
-
     auto adapt_id = [](segment_identifier const& seg_id)
     {
         return multi_and_ring_id_type{seg_id.multi_index, seg_id.ring_index};
     };
 
-    // 1 Build map of start turns (multi/ring-id -> turn indices)
-    std::map<multi_and_ring_id_type, std::vector<std::size_t>> start_turns_per_segment;
-    std::size_t index = 0;
     for (auto const& turn : turns)
     {
-        if (turn.method == method_start)
+        // Any turn which "crosses" does not have a corresponding turn.
+        // Also avoid comparing "start" with itself, or with touch/touch_interior
```

Comment:
I added this in the comment but later changed it. I’ll remove the touch part again.